lib/promscrape/config.go

```diff
@@ -433,6 +433,11 @@ func loadConfig(path string) (*Config, error) {
 	if err := c.parseData(data, path); err != nil {
 		return nil, fmt.Errorf("cannot parse Prometheus config from %q: %w", path, err)
 	}
+	// set the attach_metadata.node to true for all the kubernetes_sd_configs if
```

Comment:
Since `attach_metadata.node` is only valid for roles pod, endpoints, endpointslice, we don't need to change it for all the scraper at the beginning, 
I think we can change this when [newAPIWatcher](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/69ae1d30bf206dbf650c15370fb5ab660381bd94/lib/promscrape/discovery/kubernetes/api_watcher.go#L69C1-L69C1),
```
func newAPIWatcher(apiServer string, ac *promauth.Config, sdc *SDConfig, swcFunc ScrapeWorkConstructorFunc) (*apiWatcher, error) {
...
	selectors := sdc.Selectors
	attachNodeMetadata := sdc.AttachMetadata.Node
	if *enableAttachNodeMetadataAll && (sdc.role() == "pod" || sdc.role() == "endpoints" || sdc.role() == "endpointslice") {
		attachNodeMetadata = true
	}
```